syntax = "proto3";

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  optional string token = 2;
  optional string error_message = 3;
}

message EmbedRequest {
  string story = 1;
}

message EmbedResponse {
  repeated float embeddings = 1;
  optional string error_message = 2;
}
